src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3367,14 +3367,46 @@ public override Guid GUID
         {
             get
             {
+                TypeHandle th = GetNativeTypeHandle();
+                if (th.IsTypeDesc || IsArray)
+                {
+                    return Guid.Empty;
+                }
+
                 Guid result = default;
```

Comment:
```suggestion
                Guid result;
```
Initialization not needed